plugins/expired_gpg_keys.py

```diff
@@ -0,0 +1,105 @@
+import dnf
+import rpm
+import subprocess
+
+from datetime import datetime
+from dnfpluginscore import _, logger
+
+
+class ExpiredGPGKeys(dnf.Plugin):
+    """
+    Find expired GPG keys and suggest their removal.
+
+    This is a workaround to solve https://github.com/rpm-software-management/dnf/issues/2075.
+    """
+
+    name = 'expired-gpg-keys'
+
+    def __init__(self, base, cli):
+        super(ExpiredGPGKeys, self).__init__(base, cli)
+        self.base = base
+        self.cli = cli
+
+    def config(self):
+        if not self.base.conf.gpgcheck:
+            return
+
+        if not self.is_gpg_installed():
+            return
+
+        for (hdr, expire_date) in self.list_expired_keys():
+            print(_("The following GPG key has expired on {0}:".format(expire_date)))
+            print("    {0}\n".format(hdr["summary"]))
+            print(_("For more information about the key:"))
+            print("    rpm -qi {0}\n".format(hdr[rpm.RPMTAG_NVR]))
+
+            print(_("Installing packages signed with this key will fail due to a GPG check error.\n"
+                    "It is recommended to remove the expired key to allow the installation of\n"
+                    "packages signed with an updated key."))
+
+            if self._ask_user_no_raise(_("Do you want to remove the key?")):
+                print()
+                if self.remove_gpg_key(hdr):
+                    print(_("Key successfully removed."))
+                else:
+                    print(_("Failed to remove the key."))
+                print()
+
+    @staticmethod
+    def is_gpg_installed():
+        """
+        Check that GPG is installed to enable querying expired keys later.
+        """
+        ts = rpm.TransactionSet()
+        mi = ts.dbMatch("provides", "gpg")
+        return len(mi) > 0
+
+    @staticmethod
+    def remove_gpg_key(hdr):
+        """
+        Remove the system package corresponding to the GPG key from the given RPM header.
+        """
+        ts = rpm.TransactionSet()
+        ts.addErase(hdr)
+        error = ts.run(lambda *_: None, '')
+        return not error
+
+    @staticmethod
+    def list_expired_keys():
+        """
+        List expired GPG keys represented as `hdr` objects.
```

Comment:
The method returns different data: list of hdr-datetime tuples. I propose:

Return installed but expired PGP keys as a list of (key `hdr` object, expiration `datetime` object) tuples.